RDA and composition test failures

Master is currently failing the following RDA and composition tests:

RDA:

```shell
$ python skbio/stats/ordination/tests/test_redundancy_analysis.py
.FF
======================================================================
FAIL: test_scaling1 (__main__.TestRDAResults)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "skbio/stats/ordination/tests/test_redundancy_analysis.py", line 110, in test_scaling1
    decimal=6)
  File "/Users/jairideout/dev/scikit-bio/skbio/util/_testing.py", line 1035, in assert_ordination_results_equal
    decimal=decimal)
  File "/Users/jairideout/dev/scikit-bio/skbio/util/_testing.py", line 1083, in _assert_frame_equal
    npt.assert_almost_equal(left_values, right_values, decimal=decimal)
  File "/Users/jairideout/miniconda/envs/scikit-bio/lib/python3.5/site-packages/numpy/testing/utils.py", line 497, in assert_almost_equal
    return assert_array_almost_equal(actual, desired, decimal, err_msg)
  File "/Users/jairideout/miniconda/envs/scikit-bio/lib/python3.5/site-packages/numpy/testing/utils.py", line 892, in assert_array_almost_equal
    precision=decimal)
  File "/Users/jairideout/miniconda/envs/scikit-bio/lib/python3.5/site-packages/numpy/testing/utils.py", line 713, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Arrays are not almost equal to 6 decimals

(mismatch 31.42857142857143%)
 x: array([[ -1.209385e+00,  -9.996717e-01,   2.040803e-01,  -4.377164e-02,
          2.025459e-01,   4.174845e-02,   2.251712e-03],
       [ -1.262748e+00,  -1.114136e+00,   1.333392e-01,  -5.340052e-17,...
 y: array([[ -1.209385e+00,   9.996717e-01,   2.040803e-01,  -4.377164e-02,
         -2.025459e-01,  -4.174845e-02,   2.251712e-03],
       [ -1.262748e+00,   1.114136e+00,   1.333392e-01,  -5.659699e-17,...

======================================================================
FAIL: test_scaling2 (__main__.TestRDAResults)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "skbio/stats/ordination/tests/test_redundancy_analysis.py", line 167, in test_scaling2
    decimal=6)
  File "/Users/jairideout/dev/scikit-bio/skbio/util/_testing.py", line 1035, in assert_ordination_results_equal
    decimal=decimal)
  File "/Users/jairideout/dev/scikit-bio/skbio/util/_testing.py", line 1083, in _assert_frame_equal
    npt.assert_almost_equal(left_values, right_values, decimal=decimal)
  File "/Users/jairideout/miniconda/envs/scikit-bio/lib/python3.5/site-packages/numpy/testing/utils.py", line 497, in assert_almost_equal
    return assert_array_almost_equal(actual, desired, decimal, err_msg)
  File "/Users/jairideout/miniconda/envs/scikit-bio/lib/python3.5/site-packages/numpy/testing/utils.py", line 892, in assert_array_almost_equal
    precision=decimal)
  File "/Users/jairideout/miniconda/envs/scikit-bio/lib/python3.5/site-packages/numpy/testing/utils.py", line 713, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Arrays are not almost equal to 6 decimals

(mismatch 31.42857142857143%)
 x: array([[ -1.488490e+00,  -2.126756e+00,   7.278053e-01,  -2.272346e-01,
          3.841304e+00,   2.304877e+00,   2.600617e-01],
       [ -1.554168e+00,  -2.370273e+00,   4.755236e-01,  -2.772216e-16,...
 y: array([[ -1.488490e+00,   2.126756e+00,   7.278053e-01,  -2.272346e-01,
         -3.841304e+00,  -2.304877e+00,   2.600617e-01],
       [ -1.554168e+00,   2.370273e+00,   4.755236e-01,  -2.938157e-16,...

----------------------------------------------------------------------
Ran 3 tests in 0.194s

FAILED (failures=2)
```

Composition:

```shell
$ nosetests --with-doctest skbio/stats/composition.py
........F....
======================================================================
FAIL: Doctest: skbio.stats.composition.inner
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jairideout/miniconda/envs/scikit-bio/lib/python3.5/doctest.py", line 2189, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for skbio.stats.composition.inner
  File "/Users/jairideout/dev/scikit-bio/skbio/stats/composition.py", line 375, in inner

----------------------------------------------------------------------
File "/Users/jairideout/dev/scikit-bio/skbio/stats/composition.py", line 410, in skbio.stats.composition.inner
Failed example:
    inner(x, y)
Expected:
    0.21078524737545556
Got:
    0.21078524737545554


----------------------------------------------------------------------
Ran 13 tests in 0.063s

FAILED (failures=1)
```

The composition doctest failure should be easy to fix with `round`. The test passes with numpy 1.10.2 and fails with 1.10.4.

I'm not sure about the RDA unit test failures -- these values are different even though the tests are using `npt.assert_almost_equal`. Upgrading/downgrading scipy/numpy versions didn't resolve the issue for me as it did for the composition doctest. @Jorge-C are you available to look into the RDA tests?

--- a/skbio/stats/composition.py
+++ b/skbio/stats/composition.py
@@ -405,8 +405,8 @@ def inner(x, y):
     >>> from skbio.stats.composition import inner
     >>> x = np.array([.1, .3, .4, .2])
     >>> y = np.array([.2, .4, .2, .2])
-    >>> inner(x, y)
-    0.21078524737545556
+    >>> inner(x, y)  # doctest: +ELLIPSIS
+    0.2107852473...
     """
     x = closure(x)
     y = closure(y)

--- a/skbio/stats/ordination/tests/test_redundancy_analysis.py
+++ b/skbio/stats/ordination/tests/test_redundancy_analysis.py
@@ -106,6 +106,7 @@ class TestRDAResults(TestCase):
             eigvals=eigvals)
 
         assert_ordination_results_equal(scores, exp,
+                                        ignore_directionality=True,
                                         ignore_biplot_scores_labels=True,
                                         decimal=6)
 
@@ -163,6 +164,7 @@ class TestRDAResults(TestCase):
             eigvals=eigvals)
 
         assert_ordination_results_equal(scores, exp,
+                                        ignore_directionality=True,
                                         ignore_biplot_scores_labels=True,
                                         decimal=6)
 

--- a/skbio/util/_testing.py
+++ b/skbio/util/_testing.py
@@ -1015,7 +1015,7 @@ def assert_ordination_results_equal(left, right, ignore_method_names=False,
         Ignore differences in `biplot_scores` row and column labels.
     ignore_directionality : bool, optional
         Ignore differences in directionality (i.e., differences in signs) for
-        attributes `samples` and `features`.
+        attributes `samples`, `features` and `biplot_scores`.
 
     Raises
     ------
@@ -1042,6 +1042,7 @@ def assert_ordination_results_equal(left, right, ignore_method_names=False,
     _assert_frame_equal(left.biplot_scores, right.biplot_scores,
                         ignore_biplot_scores_labels,
                         ignore_biplot_scores_labels,
+                        ignore_directionality=ignore_directionality,
                         decimal=decimal)
 
     _assert_frame_equal(left.sample_constraints, right.sample_constraints,

--- a/skbio/util/_testing.py
+++ b/skbio/util/_testing.py
@@ -1047,6 +1047,7 @@ def assert_ordination_results_equal(left, right, ignore_method_names=False,
 
     _assert_frame_equal(left.sample_constraints, right.sample_constraints,
                         ignore_columns=ignore_axis_labels,
+                        ignore_directionality=ignore_directionality,
                         decimal=decimal)
 
     _assert_series_equal(left.eigvals, right.eigvals, ignore_axis_labels,

